File: ZemljaSlova.Model/TicketType.cs
using System;
using System.Collections.Generic;
using System.Net.Sockets;
using System.Text;

namespace ZemljaSlova.Model
{
    public class TicketType
    {
        public int Id { get; set; }

        public int EventId { get; set; }

        public decimal Price { get; set; }

        public string Name { get; set; } = null!;

        public string? Description { get; set; }

        public int? InitialQuantity { get; set; }

        public int? CurrentQuantity { get; set; }
    }
}

File: UI/zemljaslova_mobile/lib/models/ticket_type.dart
import 'package:json_annotation/json_annotation.dart';

part 'ticket_type.g.dart';

@JsonSerializable()
class TicketType {
  final int id;
  final int eventId;
  final String name;
  final double price;
  final String description;
  final int? initialQuantity;
  final int? currentQuantity;

  TicketType({
    required this.id,
    required this.eventId,
    required this.name,
    required this.price,
    required this.description,
    this.initialQuantity,
    this.currentQuantity,
  });

  factory TicketType.fromJson(Map<String, dynamic> json) => _$TicketTypeFromJson(json);

  Map<String, dynamic> toJson() => _$TicketTypeToJson(this);
} 